cherrypy/lib/httputil.py

```diff
@@ -450,12 +450,15 @@ class CaseInsensitiveDict(KeyTransformingDict):
 
     """A case-insensitive dict subclass.
 
-    Each key is changed on entry to str(key).title().
+    Each key is changed on entry to title case.
     """
 
     @staticmethod
     def transform_key(key):
-        return str(key).title()
+        if key is None:
+            # TODO: why?
+            return 'None'
```

Comment:
This should probably be
```suggestion
            raise ValueError
```

Reply:
Have you tried that? Lots of tests fail when key is `None`. I'd like to see that issue fixed too, but it's outside the scope of the reported issue.

Reply:
Oh. Then this is worth mentioning in a comment.
This should probably be
```suggestion
            # NOTE: Setting `raise ValueError` here causes many tests
            # to fail so before refactoring we'll need to fix that.
            return 'None'
```

Reply:
Right. I was trying to be more concise with `TODO: why?`... and let the failing tests tell the rest of the story until a ticket could be filed about it. With this change, we have both a TODO and a NOTE. That seems excessive, but thanks for the suggestion.

Reply:
In b07830a1, I've updated the TODO to reference a new issue (#1830).